lib/Dropdown.js

```diff
@@ -0,0 +1,775 @@
+					this.y - this.h / 2 + border / 2,
+					// Top Right
+					this.x + this.w / 2 - border / 2,
+					this.y - this.h / 2 + border / 2,
+					// Inner Top Right
+					this.x + this.w / 2 - border,
+					this.y - this.h / 2 + border,
+					// Inner Bottom Left
+					this.x - this.w / 2 + border,
+					this.y + this.h / 2 - border,
+					// Bottom Left
+					this.x - this.w / 2 + border / 2,
+					this.y + this.h / 2 - border / 2,
+				);
+                break;
+            default:
+            // Do nothing
+        }
+    }
+	/**
+     * Draws the dropdown's button according to theme.
+     */
+	#drawButton() {
+        switch (this.#pen.gui.theme) {
+            case "retro":
+				this.#pen.colour.stroke = "transparent";
+				const border = this.#borderWidth;
+
+				// Bottom-Right black bevel
+				this.#pen.colour.fill = "black";
+				const buttonWidth = this.h - border * 2;
+				const x = this.x + this.w / 2 - buttonWidth / 2 - border;
+				this.#pen.shape.rectangle(
+					x, 
+					this.y, 
+					buttonWidth, 
+					buttonWidth
+				);
+
+                // Draw a white triangle bevel on the top left
+                this.#pen.colour.fill = "white";
+                this.#pen.shape.polygon(
+                    // Top Left
+                    x - buttonWidth / 2,
+                    this.y - this.h / 2 + border,
+                    // Top Right
+                    x + buttonWidth / 2,
+                    this.y - this.h / 2 + border,
+                    // Bottom Left
+                    x - buttonWidth / 2,
+                    this.y + this.h / 2 - border,
+                );
+
+				this.#pen.colour.fill = this.background;
+				this.#pen.shape.rectangle(
+					x, 
+					this.y, 
+					buttonWidth - border * 1.5, 
+					buttonWidth - border * 1.5
+				);
+
+				
+				this.#pen.colour.fill = this.accentColour;
+
+				if (this.#openDirection === "down") {
+					// Draw down triangle
+					this.#pen.shape.polygon(
+						// Top Left
+						x - buttonWidth / 4,
+						this.y - buttonWidth / 6,
+						// Top Right
+						x + buttonWidth / 4,
+						this.y - buttonWidth / 6,
+						// Bottom
+						x,
+						this.y + buttonWidth / 6
+					);
+				} else {
+					// Draw up triangle
+					this.#pen.shape.polygon(
+						// Top Left
+						x - buttonWidth / 4,
+						this.y + buttonWidth / 6,
+						// Top Right
+						x + buttonWidth / 4,
+						this.y + buttonWidth / 6,
+						// Bottom
+						x,
+						this.y - buttonWidth / 6
+					);
+				}
+                break;
+            default:
+            // Do nothing
+        }
+    }
+	/**
+     * Draws the dropdown's options according to theme.
+     */
+	#drawOptions() {
+		const directionModifier = this.#openDirection === "up" ? -1 : 1;
+		const optionHeight = this.h - this.#borderWidth * 2;
+		const optionsTotalHeight = this.#options.length * optionHeight + this.#borderWidth * 1.5;
+		const optionsCentreY = this.y + directionModifier * (this.h / 2 + optionsTotalHeight / 2);
```

Comment:
(this.h / 2 + optionsTotalHeight / 2) can this part be given a variable name so it's clearer what it's for?